README.md

```diff
@@ -30,10 +30,11 @@ production version as necessary:
 
     gcloud config configurations create gotraining
     gcloud config set account your@email.address
-    gcloud config set project gotraining
+    gcloud config set project gotraining-testing
```

Comment:
You shouldn't need to do this `gcloud config set project ...` if you reference the project with `--project=...` when you deploy the app. But you may need to test that.

Reply:
Yes you shouldn't need to. It's a tricky one, do we get rid of the separate configuration entirely and just go with a single config linked to anz account, or do we just leave it as is and make a more prominent warning?

Reply:
it doesn't need to be an "anz account", just the same account for each project which is the most likely scenario anyway. I'm not sure why someone would use one account for `gotraining` and another for `gotraining-testing`, but if they do, I'd say it's up to them to figure out how to make it work for them and not document that here. This docco should just be the simplest way to get it working.

Reply:
yeah I'm not sure either. `gcloud` - has me confused.
personally i have a configuration for each project that I'm using and a generic `gmail` config:
```
> gcloud config configurations list
NAME                IS_ACTIVE  ACCOUNT                PROJECT              
ob                  False      julia.ogris@anz.com    anz-cx-ob-np-f67747
anzedu              False      julia.ogris@anz.com    anzedu
gmail               False      julia.ogris@gmail.com
gotraining          False      julia.ogris@anz.com    gotraining
gotraining-testing  True       julia.ogris@anz.com    gotraining-testing
```
When I switch context / repo project I'm working on I switch configurations with:
```
gcloud config configurations activate <NAME> 
```
a bit like virtualenv ;) which means I can drop the `--project`, so I just run `gcloud app deploy`.

Maybe that's stupid. I'm not sure....

Either way, I suggest either dropping the line `gcloud config set project gotraining-testing`
or renaming configuration to: `gcloud config configurations create gotraining-testing`.

Hope that's useful.

Reply:
The reason I prefer `--project=...` in the README is that it is context-independent and unambiguous. You only need to set the project once for many deployments, so I don't believe the instructions should say: "1. activate. 2. deploy". Also with `--project=...`, you can just cut and paste the one line.

Providing a set of instructions/commands that is simple and just works is important. People are free to figure out other ways of doing it if they find them more suitable to their workflow.

Configurations are only necessary when you are managing multiple contexts (usually with multiple accounts I guess), so I don't think it is necessary to document that complexity here, but a little may help.

This means the setup and deployment instructions can be as simple as:

## Setup
(if necessary)
```
gcloud config set auth/disable_ssl_validation
```

```
gcloud config set account your@email.address
gcloud auth login
```

## Deployment
You can remove `--project=gotraining-testing` from the commands below if you have run `gcloud config set project gotraining-testing`.
1. Add `service: prNN` to `app.yaml`
1. `gcloud --project=gotraining-testing app deploy`
1. ...
1. `gcloud --project=gotraining-testing app services delete prNN`